data/src/main/java/com/mashup/alcoholfree/data/repository/DataStoreRepositoryImpl.kt

```diff
@@ -0,0 +1,32 @@
+package com.mashup.alcoholfree.data.repository
+
+import androidx.datastore.core.DataStore
+import androidx.datastore.preferences.core.Preferences
+import androidx.datastore.preferences.core.edit
+import androidx.datastore.preferences.core.stringPreferencesKey
+import com.mashup.alcoholfree.domain.repository.DataStoreRepository
+import kotlinx.coroutines.flow.Flow
+import kotlinx.coroutines.flow.map
+import javax.inject.Inject
+
+class DataStoreRepositoryImpl @Inject constructor(
+    private val dataStore: DataStore<Preferences>,
+) : DataStoreRepository {
```

Comment:
```suggestion
class LoginRepositoryImpl @Inject constructor(
    private val dataStore: DataStore<Preferences>,
) : LoginRepository {
```
여기서 수행하는 역할은 로그인 관련인데 데이터스토어라고 되어있으면 나중에 서버랑 로그인 관련 로직을 수행할 때는 또 repository를 따로 만들어야할거같아
그것보다는 같은 기능을 하는거라면 하나의 repository에서 수행하는 것이 더 좋아보이네..!
어떻게 생각해?

Reply:
@AnKyungMoo 훔 형님 그러면 데이터 스토어 로직을 사용하는 로그인이나 서버 관련 로직은 DataStoreRepositoryImpl에서 함수로 만들  유스케이스로 처리 가능할 거 같고, 나중에 로그인이 필요한 로직이 들어오게되면 그때 그거를 LoginRepository로 분리하는게 낫지 않나용?

Reply:
그렇다면 나는 datasource로 하나 더 분리하는게 좋을꺼같은데 이건 어떻게 생각해?

Reply:
넹넹 형님 그게 맞는거 같아요.. 제가 제 코드를 구현한 것과 다르게 생각하고 있었네용..ㅎ 수정하고 다시 request 하겠습니다!